private/bufpkg/bufcheck/buflint/buflint_test.go

```diff
@@ -986,6 +1064,48 @@ func testLintWithModifiers(
 	)
 	require.NoError(t, err)
 
+	var validateReadWriteBucket storage.ReadWriteBucket
+	if dependencyPathPrefix != "" {
+		validateReadWriteBucket, err = storageosProvider.NewReadWriteBucket(
+			"testdata",
+			storageos.ReadWriteBucketWithSymlinksIfSupported(),
+		)
+		require.NoError(t, err)
+
+		// Copy the testdata directory to a temporary directory
+		fn := func(objectInfo storage.ObjectInfo) error {
+			get, err := validateReadWriteBucket.Get(ctx, objectInfo.Path())
+			if err != nil {
+				return err
+			}
+			defer get.Close()
+			expectedData, err := io.ReadAll(get)
+			if err != nil {
+				return err
+			}
+			put, err := readWriteBucket.Put(ctx, objectInfo.Path())
+			if err != nil {
+				return err
+			}
+			defer put.Close()
+			_, err = put.Write(expectedData)
+			if err != nil {
+				return err
+			}
+			return nil
+		}
+		if err := validateReadWriteBucket.Walk(ctx, dependencyPathPrefix, fn); err != nil {
```

Comment:
This all would be replaced with `storage.Copy`, but as mentioned, I don't think we need to (or should be) copying any data.